include/nbl/video/alloc/SubAllocatedDescriptorSet.h

```diff
@@ -23,26 +23,66 @@ class SubAllocatedDescriptorSet : public core::IReferenceCounted
 	using value_type = typename AddressAllocator::size_type;
 	static constexpr value_type invalid_value = AddressAllocator::invalid_address;
 
+	class DeferredFreeFunctor
+	{
+	public:
+		inline DeferredFreeFunctor(SubAllocatedDescriptorSet* composed, uint32_t binding, size_type count, value_type* addresses)
+			: m_addresses(addresses, addresses + count), m_binding(binding), m_composed(composed)
+		{
+		}
+
+		// Just does the de-allocation
+		inline void operator()()
+		{
+			// isn't assert already debug-only?
```

Comment:
yes, but Erfan wanted to change our debugbreaks and asserts a bit #352 so doing them like that for extra clarity